lib/object_3d.dart

```diff
@@ -45,6 +53,29 @@ class _Object3DState extends State<Object3D> {
       // Load the object from a string
       _parseObj(widget.object!);
     }
+
+    final double maxSpeed = widget.maxSpeed.abs();
+    final double dampCoef = widget.dampCoef.clamp(0.001, 0.999);
+
+    _updateTimer = Timer.periodic(Duration(milliseconds: 16), (_) {
+      if (!mounted) return;
+      setState(() {
+        final double adx = _deltaX.abs();
+        final double ady = _deltaY.abs();
+        final double sx = _deltaX < 0 ? -1 : 1;
+        final double sy = _deltaY < 0 ? -1 : 1;
+
+        _deltaX = math.min(maxSpeed, adx) * sx;
+        _deltaY = math.min(maxSpeed, ady) * sy;
+
+        _deltaX *= dampCoef;
+        _deltaY *= dampCoef;
```

Comment:
These lines can also be combined immediately to make it a bit shorter

```suggestion
        _deltaX = math.min(maxSpeed, adx) * sx * dampCoef;
        _deltaY = math.min(maxSpeed, ady) * sy * dampCoef;
```